lib/inspec/plugins/resource.rb

```diff
@@ -37,14 +37,24 @@ def __resource_registry
       Inspec::Resource.registry
     end
 
-    def __register(name, obj)
+    def __register(name, obj) # rubocop:disable Metrics/MethodLength
       cl = Class.new(obj) do
         def initialize(backend, name, *args)
+          @resource_skipped = nil
+          @resource_failed = nil
+
           # attach the backend to this instance
           @__backend_runner__ = backend
           @__resource_name__ = name
+
           # call the resource initializer
-          super(*args)
+          begin
+            super(*args)
+          rescue Inspec::Exceptions::ResourceSkipped => e
+            skip_resource(e.message)
+          rescue Inspec::Exceptions::ResourceFailed => e
+            fail_resource(e.message)
+          end
```

Comment:
What happens in cases where we have an unhanded exception. I think we should handle those cases as well.

Reply:
Ugh, can't believe I missed this. An `else raise` is totally appropriate here.

Reply:
I take it back... Ruby will raise an exception if it's not one of the rescued classes. `else raise` is implicit. Given that we would just bubble up an exception anyways without this change (i.e. syntax issues in the initialize method), I think this is fine as-is.

Reply:
In other words, we currently unwind InSpec if there's a ruby syntax issue inside a resource. We should continue to do that in my opinion.